code/modules/mob/living/simple_mob/subtypes/mechanical/hivebot/support.dm

```diff
@@ -0,0 +1,87 @@
+// These hivebots help their team in various ways, and can be very powerful with allies, but are otherwise very weak when alone.
+
+/mob/living/simple_mob/mechanical/hivebot/support
+	icon_state = "white"
+	icon_living = "white"
+	attacktext = list("prodded")
+	movement_cooldown = 5
+	melee_damage_lower = 2
+	melee_damage_upper = 2
+
+
+// This hivebot supplies a general buff to nearby hivebots that improve their performance.
+// Note that the commander itself does not receive the buff.
+/mob/living/simple_mob/mechanical/hivebot/support/commander
+	name = "commander hivebot"
+	desc = "A robot that appears to be directing the others."
+	maxHealth = 5 LASERS_TO_KILL // 150 health
+	health = 5 LASERS_TO_KILL
+	player_msg = "You <b>increase the performance of other hivebots near you</b> passively.<br>\
+	You are otherwise very weak offensively."
+
+/mob/living/simple_mob/mechanical/hivebot/support/commander/handle_special()
+	for(var/mob/living/L in range(4, src))
+		if(L == src)
+			continue // Don't buff ourselves.
+		if(L.faction == src.faction && L.isSynthetic()) // Don't buff enemies.
```

Comment:
Faction is a string, e.g. "hivebot" or "carp".

There is a list but its held inside the AI and is used for calling for help and is currently not 101% reliable due to VV not updating it. Fixing that requires rewriting the faction system, writing a function to change factions, and modernizing our View-Variables to be able to use a specific function for a specific variable (like TG has iirc) instead of just hard-setting a var, which seemed out of scope for the AI rewrite.

The list also being in the AI means that the list might not exist if the mob does not have an AI datum attached for whatever reason, so using that list wouldn't be good.

Instead, the 'is this guy a good lad' interface function, ``IIsAlly(mob)`` would be better since it checks for other things besides faction equality.